### PR TITLE
interface version 6, includes new bridge and context

### PR DESCRIPTION
**Pull request by GalSvirin**

### CODE REVIEW

File: takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/core/contexts/TakipiAgentP6Context.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.takipi.sdk.v1.internal.impl.agent.p6.core.contexts;
+
+import com.takipi.sdk.v1.internal.agent.shared.p1.core.contexts.TakipiInternalP1Context;
+import com.takipi.sdk.v1.internal.impl.agent.p1.core.contexts.TakipiAgentP1Context;
+
+public class TakipiAgentP6Context extends TakipiAgentP1Context
+{
+	protected TakipiAgentP6Context(Class<?> clazz, String methodName, String path, TakipiInternalP1Context internalContext)
```
Comment (M-Tsur) [inline takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/core/contexts/TakipiAgentP6Context.java]: This is ctor is unused and actually should extend TakipiAgentP5Context. I don't think we actually need it

File: takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	
+	private final TakipiInternalP6AgentBridge bridge;
+	
+	protected TakipiAgentP6ClientImpl(String frameworkId, String clientId, TakipiOptions options, TakipiInternalP6AgentBridge bridge)
+	{
+		super(frameworkId, clientId, options, bridge);
+		this.bridge = bridge;
+	}
+	
+	public TakipiContext createInternalContext(Class<?> clazz, String methodName, String methodDesc)
```
Comment (M-Tsur) [inline takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java]: why do we have createInternalContext as a public function here? is it really needed?
Other clients have createContext and nothing has to do with it being internal

**Review by M-Tsur — CHANGES_REQUESTED** [review]

File: takipi-sdk-agent-shared/src/main/java/com/takipi/sdk/v1/internal/agent/shared/p6/bridge/TakipiInternalP6AgentBridge.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.takipi.sdk.v1.internal.agent.shared.p6.bridge;
+
+import com.takipi.sdk.v1.internal.agent.shared.p5.bridge.TakipiInternalP5AgentBridge;
+import com.takipi.sdk.v1.internal.agent.shared.p5.core.contexts.TakipiInternalP5Context;
+
+public interface TakipiInternalP6AgentBridge extends TakipiInternalP5AgentBridge
+{
```
Comment (nivstein) [inline takipi-sdk-agent-shared/src/main/java/com/takipi/sdk/v1/internal/agent/shared/p6/bridge/TakipiInternalP6AgentBridge.java]: Need to keep standard Java formatting in the public repo (i.e. curly braces on same line as declaration)

File: takipi-sdk-agent-shared/src/main/java/com/takipi/sdk/v1/internal/agent/shared/p6/bridge/TakipiInternalP6AgentBridge.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+	public TakipiInternalP5Context createInternalContext(Class<?> clazz, String methodName, String methodDesc);
+	public TakipiInternalP5Context createInternalContext(long contextHash);
```
Comment (nivstein) [inline takipi-sdk-agent-shared/src/main/java/com/takipi/sdk/v1/internal/agent/shared/p6/bridge/TakipiInternalP6AgentBridge.java]: There two methods should return a P6 context.

File: takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import com.takipi.sdk.v1.api.TakipiOptions;
+import com.takipi.sdk.v1.api.core.contexts.TakipiContext;
+import com.takipi.sdk.v1.internal.TakipiClient;
+import com.takipi.sdk.v1.internal.agent.shared.p6.bridge.TakipiInternalP6AgentBridge;
+import com.takipi.sdk.v1.internal.impl.agent.TakipiAgentInterface;
+import com.takipi.sdk.v1.internal.impl.agent.p5.TakipiAgentP5ClientImpl;
+import com.takipi.sdk.v1.internal.impl.agent.p5.core.contexts.TakipiAgentP5Context;
+
+public class TakipiAgentP6ClientImpl extends TakipiAgentP5ClientImpl
+{
```
Comment (nivstein) [inline takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java]: Curly braces on line above.

File: takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+	public static TakipiClient create(String frameworkId, String clientId, TakipiOptions options) {
+		return new TakipiAgentP6ClientImpl(frameworkId, clientId, options,
+				TakipiAgentInterface.create().createP6Bridge(frameworkId, clientId));
+	}
+	
+	private final TakipiInternalP6AgentBridge bridge;
+	
+	protected TakipiAgentP6ClientImpl(String frameworkId, String clientId, TakipiOptions options, TakipiInternalP6AgentBridge bridge)
+	{
```
Comment (nivstein) [inline takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java]: Curly

File: takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+public class TakipiAgentP6ClientImpl extends TakipiAgentP5ClientImpl
+{
+	public static TakipiClient create(String frameworkId, String clientId, TakipiOptions options) {
+		return new TakipiAgentP6ClientImpl(frameworkId, clientId, options,
+				TakipiAgentInterface.create().createP6Bridge(frameworkId, clientId));
+	}
+	
+	private final TakipiInternalP6AgentBridge bridge;
+	
+	protected TakipiAgentP6ClientImpl(String frameworkId, String clientId, TakipiOptions options, TakipiInternalP6AgentBridge bridge)
```
Comment (nivstein) [inline takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java]: Why is this `protected` and not `private`?

File: takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+		
+		return TakipiAgentP5Context.create(clazz, path, bridge.createInternalContext(clazz, path));
+	}
+	
+	public TakipiContext createInternalContext(long contextHash)
+	{
+		String path = String.valueOf(contextHash);
+		
+		return TakipiAgentP5Context.create(Object.class, path, bridge.createInternalContext(Object.class, path));
+	}
```
Comment (nivstein) [inline takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java]: Where are the `@Overrides` of the actual `createContext`s?

File: takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/core/contexts/TakipiAgentP6Context.java (hunk 1 above)
Comment (nivstein) [reply takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/core/contexts/TakipiAgentP6Context.java]: Agreed

File: takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java (hunk 2 above)
Comment (nivstein) [reply takipi-sdk/src/main/java/com/takipi/sdk/v1/internal/impl/agent/p6/TakipiAgentP6ClientImpl.java]: Agreed

**Review by nivstein — CHANGES_REQUESTED** [review]

**Review by M-Tsur — APPROVED** [review]

File: takipi-sdk-agent-shared/src/main/java/com/takipi/sdk/v1/internal/agent/shared/p6/bridge/TakipiInternalP6AgentBridge.java [hunk 9]
```diff
@@ -3,9 +3,9 @@
 import com.takipi.sdk.v1.internal.agent.shared.p5.bridge.TakipiInternalP5AgentBridge;
 import com.takipi.sdk.v1.internal.agent.shared.p5.core.contexts.TakipiInternalP5Context;
 
-public interface TakipiInternalP6AgentBridge extends TakipiInternalP5AgentBridge
-{
+public interface TakipiInternalP6AgentBridge extends TakipiInternalP5AgentBridge {
+	
 	public TakipiInternalP5Context createInternalContext(Class<?> clazz, String methodName, String methodDesc);
-	public TakipiInternalP5Context createInternalContext(long contextHash);
+	public TakipiInternalP5Context createInternalContext(String ClassName, String methodName, String methodDesc);
```
Comment (nivstein) [inline takipi-sdk-agent-shared/src/main/java/com/takipi/sdk/v1/internal/agent/shared/p6/bridge/TakipiInternalP6AgentBridge.java]: `ClassName` -> `className`

**Review by nivstein — CHANGES_REQUESTED** [review]

**Pull request merged**